packages/library-base/src/commonMain/kotlin/io/realm/kotlin/types/StaticRealmObject.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.types
+
+/**
+ * Base interface for all statically typed realm classes.
+ */
+public interface StaticRealmObject : BaseRealmObject
```

Comment:
Yeah, I was not really sure about the name, I considered `TypedRealmObject` as well... but because `DynamicRealmObject` is a dynamically typed object then the other is a statically typed object.

I don't have any strong feelings about the name though. 

It might feel a bit much to add this interface, but it saves me writing an extra linking objects delegate for embedded objects sources.